src/types/index.ts

```diff
@@ -239,3 +239,21 @@ export type SaveMutationInput = {
   id: string[];
   timestamp: Date;
 };
+
+export type SaveUpdateTagsInputGraphql = {
+  saveId: string;
+  removeTagIds: string[];
+  addTagNames: string[];
+};
+
+/**
+ * Input in a better format for doing bulk transactions. Group
+ * into array of creates and deletes.
+ * Transform the tagIds into "names", which corresponds to the
+ * item_tags.tag field in the database and is part of the primary key
+ * (which also includes user_id and item_id).
```

Comment:
user_id included only bc of the user_id + item_id == savedItemId approach, right? (just trying to map the definition to what I see defined here)

Reply:
Thank you, this comment is worded poorly. What I was trying to say is that the item_tags primary key is `(tag, user_id, item_id)`. So transform the input to extract the tag name from the "id" input (which at this time is constructed from the tag name `item_tags.tag`, since there isn't a concept of a tag id.)